.github/workflows/build-and-test.yml

```diff
@@ -1,415 +1,540 @@
+          requirements-files: requirements.txt
+          requirements: ${{ env.DEFAULT_DEPENDENCIES }} pytest
+
+      - name: Make install
+        run: make install
+
+      - name: Validate sample data
+        run: make validate_sampledata
+
+      - name: Validate user guide links
+        run: make validate_userguide_filenames
+
+  run-ui-tests:
+    name: Run UI Tests
+    runs-on: windows-latest
+    needs: check-syntax-errors
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # Fetch complete history for accurate versioning
+
+      - name: Restore git-LFS test data cache
+        uses: actions/cache@v2
+        with:
+          path: data/invest-test-data
+          key: git-lfs-testdata-${{ hashfiles('Makefile') }}
+
+      - uses: ./.github/actions/setup_env
+        with:
+          python-version: ${{ matrix.python-version }}
+          requirements-files: requirements.txt requirements-dev.txt requirements-gui.txt
+          requirements: ${{ env.DEFAULT_DEPENDENCIES }}
+
+      - name: Make install
+        run: make install
+
+      - name: Run UI tests
+        timeout-minutes: 10  # tests usually take < 2 minutes, so 10 is generous.
+        run: make test_ui
+
+  run-workbench-tests:
+    name: Run Workbench Tests
+    runs-on: ${{ matrix.os }}
+    needs: check-syntax-errors
+    strategy:
+      fail-fast: false
+      max-parallel: 4
+      matrix:
+        os: [windows-latest, macos-latest]
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # Fetch complete history for accurate versioning
+
+      - name: Set up python environment
+        uses: ./.github/actions/setup_env
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+          requirements-files: requirements.txt requirements-dev.txt requirements-gui.txt
+          requirements: ${{ env.DEFAULT_DEPENDENCIES }}
+
+      - name: Make install
+        run: make install
+
+      - name: Set up node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14.x
+
+      - name: Restore node_modules cache
+        uses: actions/cache@v2
+        with:
+          path: workbench/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('workbench/yarn.lock') }}
+
+      - name: Install workbench dependencies
+        working-directory: workbench
+        run: |
+          yarn config set network-timeout 600000 -g
+          yarn install
+
+      - name: Run workbench tests
+        working-directory: workbench
+        env:
+          CI: true
+        run: yarn test
+
+  build-binaries:
+    name: Build binaries
+    needs: check-syntax-errors
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-10.15, windows-latest]
+        include:  # test with a range of numpy and gdal versions

```

Comment:
I think we're no longer testing with a range of numpy and GDAL versions here, right?  Just building the binaries?

Reply:
ah right, I removed this comment.